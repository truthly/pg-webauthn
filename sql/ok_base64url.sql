BEGIN;

CREATE EXTENSION webauthn CASCADE;

SET search_path TO webauthn, public, pg_temp;

SELECT webauthn.base64url_decode(webauthn.base64url_encode(x)) = x FROM gen_random_bytes(1) AS x;
SELECT webauthn.base64url_decode(webauthn.base64url_encode(x)) = x FROM gen_random_bytes(10) AS x;
SELECT webauthn.base64url_decode(webauthn.base64url_encode(x)) = x FROM gen_random_bytes(100) AS x;
SELECT webauthn.base64url_decode(webauthn.base64url_encode(x)) = x FROM gen_random_bytes(1000) AS x;

ROLLBACK;
