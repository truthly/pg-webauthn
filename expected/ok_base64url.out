BEGIN;
CREATE EXTENSION webauthn CASCADE;
NOTICE:  installing required extension "pguecc"
NOTICE:  installing required extension "pgcrypto"
SET search_path TO webauthn, public, pg_temp;
SELECT webauthn.base64url_decode(webauthn.base64url_encode(x)) = x FROM gen_random_bytes(1) AS x;
 ?column? 
----------
 t
(1 row)

SELECT webauthn.base64url_decode(webauthn.base64url_encode(x)) = x FROM gen_random_bytes(10) AS x;
 ?column? 
----------
 t
(1 row)

SELECT webauthn.base64url_decode(webauthn.base64url_encode(x)) = x FROM gen_random_bytes(100) AS x;
 ?column? 
----------
 t
(1 row)

SELECT webauthn.base64url_decode(webauthn.base64url_encode(x)) = x FROM gen_random_bytes(1000) AS x;
 ?column? 
----------
 t
(1 row)

ROLLBACK;
